sql/expression/function/date.go

```diff
@@ -456,7 +543,10 @@ func (ut *UnixTimestamp) IsNullable() bool {
 }
 
 func (ut *UnixTimestamp) Type() sql.Type {
-	return types.Float64
+	if ut.typ == nil {
```

Comment:
can type be nil? with the constructor changes?

Reply:
The changes do add either `Int64` or `Float64`, but I don't think it's impossible for type to the `nil`.
The check will prevent panics.